### PR TITLE
TF-M: Fix Kconfig structure for passing TFM_PROFILE and AUDIT_LOG configurations

### PR DESCRIPTION
**Pull request by ioannisg**
This PR contains the proper fix for enabling/disabling the Audit Log service.
- The original workaround commit is reverted (@galak FYI. I recall you had a question about this issue, whether this is an upstream bug)
- The Audit Log may not be enabled, now, together with the IPC model; the require dependencies are added. See the details here: https://developer.trustedfirmware.org/T892 for the rationale
- The TFM profile is now a choice with options 
    - none, small, medium large (Large profile is new in TFM 1.3, so we also update the Kconfig symbol docs)
    - different Kconfig TFM profile options select respective TFM Kconfig configuration, to reflect what happens under the hood in 
       the TFM Cmake.
    - the TFM_PROFILE is now a hidden symbol, so it is not selected directly by the users.
      The NXP xpresso platform is selecting the choice symbol for medium profile.

### CODE REVIEW

**Comment by ioannisg** [comment]
@galak this should address your question regarding the audit log service in the PR we merged earlier.

File: boards/arm/lpcxpresso55s69/Kconfig.defconfig [hunk 1]
```diff
@@ -53,7 +53,7 @@ config FLASH_LOAD_SIZE
 endif # TRUSTED_EXECUTION_NONSECURE || BOARD_LPCXPRESSO55S69_CPU1
 
 config TFM_PROFILE
```
Comment (oyvindronningstad) [inline boards/arm/lpcxpresso55s69/Kconfig.defconfig:55]: Should be TFM_PROFILE_TYPE

Comment (ioannisg) [reply boards/arm/lpcxpresso55s69/Kconfig.defconfig]: Fixed, thanks.

**Comment by ioannisg** [comment]
@microbuilder @galak could i get your reviews here?

File: modules/trusted-firmware-m/Kconfig [hunk 2]
```diff
@@ -64,12 +64,41 @@ config TFM_KEY_FILE_NS
...
-	  to have this configuration set.
+	  profile_large, profile_medium and profile_small. The default profile
+	  does not need to have this configuration set.
+
+choice TFM_PROFILE_TYPE
+	prompt "TF-M build profile"
+	depends on BUILD_WITH_TFM
+	default TFM_PROFILE_TYPE_NOT_SET
+	help
+	  The TF-M build profile selection. Can be  empty (not set),
```
Comment (microbuilder) [inline modules/trusted-firmware-m/Kconfig:82]: nitpick, but double space
```suggestion
	  The TF-M build profile selection. Can be empty (not set),
```

**Review by microbuilder — CHANGES_REQUESTED** [review]

File: modules/trusted-firmware-m/Kconfig (hunk 2 above)
Comment (ioannisg) [reply modules/trusted-firmware-m/Kconfig]: fixed, thanks!

**Review by microbuilder — APPROVED** [review]

File: modules/trusted-firmware-m/Kconfig [hunk 3]
```diff
@@ -96,6 +125,9 @@ config TFM_ISOLATION_LEVEL
 	int "Isolation level setting."
 	range 1 3
 	depends on BUILD_WITH_TFM
+	default 1 if TFM_PROFILE_TYPE_SMALL
```
Comment (oyvindronningstad) [inline modules/trusted-firmware-m/Kconfig:128]: Note that these do not update if TFM_PROFILE_TYPE is updated via menuconfig.

File: modules/trusted-firmware-m/Kconfig [hunk 4]
```diff
@@ -64,12 +64,41 @@ config TFM_KEY_FILE_NS
...
+
+config TFM_PROFILE_TYPE_NOT_SET
+	bool "TF-M build profile is not set"
+
+config TFM_PROFILE_TYPE_SMALL
+	bool "TF-M build profile: small"
+
+config TFM_PROFILE_TYPE_MEDIUM
+	bool "TF-M build profile: medium"
+	select TFM_IPC
```
Comment (oyvindronningstad) [inline modules/trusted-firmware-m/Kconfig:95]: I think I'd prefer if this was handled in the TFM_IPC config like for the ISOLATION_LEVEL. I think it's clearer that way, and if we later make IPC the default, profile_small will still have the correct TFM_IPC setting.

File: modules/trusted-firmware-m/Kconfig [hunk 5]
```diff
@@ -64,12 +64,41 @@ config TFM_KEY_FILE_NS
...
+	string
 	depends on BUILD_WITH_TFM
+	default "profile_small" if TFM_PROFILE_TYPE_SMALL
+	default "profile_medium" if TFM_PROFILE_TYPE_MEDIUM
+	default "profile_large" if TFM_PROFILE_TYPE_LARGE
 	help
 	  Build profile used to build tfm_s image. The available values are
-	  profile_medium and profile_small. The default profile does not need
-	  to have this configuration set.
+	  profile_large, profile_medium and profile_small. The default profile
```
Comment (oyvindronningstad) [inline modules/trusted-firmware-m/Kconfig:74]: Setting this config via menuconfig doesn't really work well. Should that be mentioned here?

File: modules/trusted-firmware-m/Kconfig (hunk 4 above)
Comment (ioannisg) [reply modules/trusted-firmware-m/Kconfig]: done.

File: modules/trusted-firmware-m/Kconfig (hunk 3 above)
Comment (ioannisg) [reply modules/trusted-firmware-m/Kconfig:126]: Yes, but we should not care, right? I made the option hidden when the profile symbol is set, anyways, to reflect that the TFM will configure its own isolation level and the symbols choice, here, won't matter.

File: modules/trusted-firmware-m/Kconfig [hunk 6]
```diff
@@ -93,12 +120,17 @@ config TFM_CMAKE_BUILD_TYPE_DEBUG
 endchoice
 
 config TFM_ISOLATION_LEVEL
-	int "Isolation level setting."
+	int "Isolation level setting." if !(TFM_PROFILE_TYPE_SMALL || TFM_PROFILE_TYPE_MEDIUM || TFM_PROFILE_TYPE_LARGE)
```
Comment (oyvindronningstad) [inline modules/trusted-firmware-m/Kconfig:123]:
```suggestion
	int "Isolation level setting." if TFM_PROFILE_TYPE_NOT_SET
```

File: modules/trusted-firmware-m/Kconfig [hunk 7]
```diff
@@ -172,24 +204,28 @@ config TFM_PARTITION_PLATFORM
...
 	  in the trusted-firmware-m repository for details regarding this
 	  parameter. Any dependencies between the various TFM_PARTITION_*
 	  options are handled by the build system in the trusted-firmware-m
 	  repository.
+	  Note: the Audit Log service does not implement the IPC model
+	  interface so it may not be enabled together with IPC option.
 
 config TFM_IPC
-	bool "IPC"
+	bool "IPC" if !(TFM_PROFILE_TYPE_SMALL || TFM_PROFILE_TYPE_MEDIUM || TFM_PROFILE_TYPE_LARGE)
```
Comment (oyvindronningstad) [inline modules/trusted-firmware-m/Kconfig:222]:
```suggestion
	bool "IPC" if TFM_PROFILE_TYPE_NOT_SET
```

File: modules/trusted-firmware-m/Kconfig (hunk 5 above)
Comment (oyvindronningstad) [reply modules/trusted-firmware-m/Kconfig:74]: With the conditional removal of the prompt, i don't think we need to mention anything.

File: modules/trusted-firmware-m/Kconfig (hunk 6 above)
Comment (ioannisg) [reply modules/trusted-firmware-m/Kconfig:123]: Sure, I can do that. :)
I thought it was a bit more pedantic if I list all the existing profiles.
Will apply your fb now.

Comment (ioannisg) [reply modules/trusted-firmware-m/Kconfig]: done

**Review by oyvindronningstad — APPROVED** [review]

File: modules/trusted-firmware-m/Kconfig (hunk 7 above)
Comment (ioannisg) [reply modules/trusted-firmware-m/Kconfig]: done

**Pull request merged**